### PR TITLE
Reject Oversized Scalar Tid Launches

### PR DESCRIPTION
**Pull request by shi-eric**
## Description

Prevent scalar `wp.tid()` coordinates from wrapping when a kernel's retained one-dimensional launch extent exceeds `2**31`. These launches now raise a `ValueError` before dispatch, including recorded launches and JAX lowering, while multidimensional and no-`tid` launches retain their existing behavior.

Scalar `wp.tid()` returns a signed 32-bit coordinate, but the host previously accepted extents whose coordinates exceeded that range. The new limit is derived during kernel reference analysis and ignores calls removed by static control flow, so validation reflects the generated kernel body rather than source syntax.

APIC graph save/load behavior for otherwise valid folded bounds is tracked separately in [GH-1800](https://github.com/NVIDIA/warp/issues/1800). In particular, this change does not alter APIC serialization of `coord_mult` for `dim=(2**31, 2)`.

## Changes

- Record whether generated kernel code uses scalar `wp.tid()` and validate its retained launch extent when building or resizing launch bounds.
- Apply the same validation to direct launches, recorded launches, JAX FFI, and legacy JAX custom calls.
- Remove the native post-dispatch warning, add a no-`tid` launch benchmark, and add a changelog fragment.
- Cover the signed 32-bit boundary, folded dimensions, mixed `wp.tid()` arities, static-dead calls, recorded command updates, and JAX tracing.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/warp/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] I added a changelog fragment if this change affects users.

## Validation summary

- Verified the static-dead `wp.tid()` regression test failed before the reachability fix and passes afterward.
- Ran all 76 launch-bounds tests on CPU and two CUDA devices.
- Ran the oversized-dimension JAX tests with current JAX and JAX 0.7.2, covering FFI and legacy lowering.
- Ran the full Warp suite: 15,267 tests passed with 32 expected skips.
- Ran all applicable pre-commit hooks.

## Bug fix

Before this change, the recorded launch below was accepted even though dispatch could produce scalar thread coordinates outside the signed 32-bit range. It now raises a `ValueError` before dispatch.

```python
import warp as wp


@wp.kernel
def scalar_tid_kernel():
    i = wp.tid()


wp.launch(
    scalar_tid_kernel,
    dim=2**31 + 1,
    device="cpu",
    record_cmd=True,
)
```

## Summary by CodeRabbit

* **Bug Fixes**
  * One-dimensional kernels using scalar `wp.tid()` now reject launch extents larger than `2**31` with a clear `ValueError`, preventing thread-coordinate overflow.
  * Validation is consistent across standard, JAX custom-call, and FFI launches, including recorded launch dimension updates.
  * Oversized launches remain supported when scalar `wp.tid()` usage is absent or removed during compilation.
* **Documentation**
  * Documented the scalar thread-index launch limit and error behavior.
* **Tests**
  * Added coverage for boundary conditions, aliases, dead code, multidimensional launches, and JAX integrations.

### CODE REVIEW

File: warp/tests/interop/test_jax.py [hunk 1]
```diff
@@ -522,6 +530,123 @@ def f_2d():
...
+
+    @jax.jit
+    def run():
+        return jax_inc(jp.ones(1, dtype=jp.float32))
+
+    with jax.default_device(wp.device_to_jax(device)):
+        with test.assertRaisesRegex(
+            ValueError, r"Warp cannot launch a kernel using scalar wp\.tid\(\) with extent 2147483649"
+        ):
+            run.lower()
```
[the remark this answers is not in the document]
Comment (shi-eric) [reply warp/tests/interop/test_jax.py:569]: @coderabbitai These tests are registered only inside the else branch of the module-level `try: import jax` block, which binds `jax` globally. If the import fails, the tests are not registered. I verified all three affected tests with `jax[cuda13]` on CPU and both CUDA devices; all nine cases pass, so no local binding is required.
